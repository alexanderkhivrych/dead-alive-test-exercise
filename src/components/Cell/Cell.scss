.cell {
  display: block;
  width: 10px;
  height: 10px;
  &--dead {
    background: white;
  }
  &--alive {
    background: green;
  }
}
