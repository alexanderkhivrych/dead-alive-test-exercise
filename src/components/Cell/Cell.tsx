import React from 'react';

import './Cell.scss';

type Props = {
  isAlive?: boolean;
};

const Cell: React.FC<Props> = ({ isAlive }) => <div className={`cell cell--${isAlive ? 'alive' : 'dead'}`} />;

export default Cell;
