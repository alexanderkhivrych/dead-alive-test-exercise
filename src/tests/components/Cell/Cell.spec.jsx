import React from 'react';
import { render } from '@testing-library/react';
import Cell from '../../../components/Cell';

describe('<Cell />', () => {
  test('should render cell with alive class if isAlive is true', () => {
    const { container } = render(<Cell isAlive={true} />);
    expect(container).toMatchSnapshot();
  });

  test('should render white if isAlive prop is false', () => {
    const { container } = render(<Cell />);
    expect(container).toMatchSnapshot();
  });

  test('should render white if isAlive prop is false', () => {
    const { container } = render(<Cell isAlive={false} />);
    expect(container).toMatchSnapshot();
  });
});
