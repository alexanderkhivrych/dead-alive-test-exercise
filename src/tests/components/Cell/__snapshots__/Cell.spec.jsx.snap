// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> should render cell with alive class if isAlive is true 1`] = `
<div>
  <div
    class="cell cell--alive"
  />
</div>
`;

exports[`<Cell /> should render white if isAlive prop is false 1`] = `
<div>
  <div
    class="cell cell--dead"
  />
</div>
`;

exports[`<Cell /> should render white if isAlive prop is false 2`] = `
<div>
  <div
    class="cell cell--dead"
  />
</div>
`;
